test/integration/stats_integration_test.cc

```diff
@@ -175,21 +175,29 @@ TEST_P(ClusterMemoryUtilization, MemoryLargeClusterSizeWithStats) {
   if (!Stats::TestUtil::hasDeterministicMallocStats()) {
     return;
   }
-  const size_t start_mem = Memory::Stats::totalCurrentlyAllocated() / 1000;
+  const size_t start_mem = Memory::Stats::totalCurrentlyAllocated();
 
   auto IpVersions = testing::TestWithParam<Network::Address::IpVersion>::GetParam();
 
   auto t1 = std::make_unique<BaseIntegrationTest>(IpVersions);
   t1->initialize();
   const size_t m1 = Memory::Stats::totalCurrentlyAllocated();
-  EXPECT_LT(start_mem, m1 / 1000);
-  EXPECT_LT(m1 / 1000, 3700); // actual value: 3664 as of Feb 26, 2019
+  EXPECT_LT(start_mem, m1);
   t1.reset(nullptr);
 
   const size_t m1001 = memoryConsumedWithClusters(1001, true);
-  EXPECT_LT(start_mem / 1000, m1001 / 1000);
+  EXPECT_LT(start_mem, m1001);
   size_t m_per_cluster = (m1001 - m1) / 1000;
-  EXPECT_LT(m_per_cluster, 58000); // actual value: 57936 as of Feb 26, 2019
+  // Make sure we don't regress. Data as of 2019/02/27:
+  //
+  // libstdc++:
+  // ----------
+  // m_per_cluster: 56119
+  //
+  // libc++:
+  // -------
+  // m_per_cluster: 52249
+  EXPECT_LT(m_per_cluster, 58000);
```

Comment:
Is it possible for us to detect in the code whether we are running in libstdc++ or libc++ mode? @lizan ?

It'd be good to check against the appropriate bounds, so we don't bloat the libc++ version against the libstdc++ threshold.

This could be a TODO. I think one way to do this might be add a -DUSING_LIBCPP define in the bazel infrastructure, if there's not already such a thing.

Reply:
`#ifdef _LIBCPP_VERSION`

Reply:
Great. Let's use it then in this PR.

Also can we have slightly tighter bounds? Maybe just round up to the next thousand, say 53000 for libc++ and 57000 for libstdc+++?